projects/components/src/navigation/nav-item/nav-item.component.ts

```diff
@@ -54,12 +68,46 @@ export class NavItemComponent {
   @Input()
   public collapsed: boolean = true;
 
-  public buildNavigationParam = (item: NavItemLinkConfig): NavigationParams => ({
-    navType: NavigationParamsType.InApp,
-    path: item.matchPaths[0],
-    relativeTo: this.activatedRoute,
-    replaceCurrentHistory: item.replaceCurrentHistory
-  });
+  public constructor(
+    private readonly activatedRoute: ActivatedRoute,
+    private readonly timeRangeService: TimeRangeService,
+    private readonly navigationService: NavigationService,
+    private readonly pageTimeRangeService: PageTimeRangeService
+  ) {}
+
+  public ngOnInit(): void {
+    this.navItemParams$ = this.pageTimeRangeService.getPageTimeRange(this.config.matchPaths[0]).pipe(
+      map(timeRange => {
+        if (isNil(timeRange)) {
+          this.pageTimeRangeService.setPageTimeRange(this.config.matchPaths[0], this.getDefaultPageTimeRange());
```

Comment:
so, `.setPageTimeRange` is just storing the default in preference service?

Reply:
hmm. Imo, the `pageTimeRangeService`  should be responsible for maintaining `getDefaultPageTimeRange()` logic and it should return default TR if timerange is not defined in pref service..

Reply:
okay, I thought because it has more context about the current location it would serve better for future changes if needed. However now that you mention it I think you're right. If in the future we want to move to saving time ranges for non-first level routes then it should definitely be in the service. And it's logic has more to do with the the service than it does the nav-item.